src/System.Private.CoreLib/src/System/Globalization/FormatProvider.FormatAndParse.Unix.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime;
+using System.Globalization;
+using System.Diagnostics;
+
+namespace System.Globalization
+{
+    internal partial class FormatProvider
+    {
+        private partial class Number
+        {
+            // buffer size to hold digits (ECVT_MAX_COUNT_SIZE), decimal point, number sign, exponent, exponent symbol 'e', exponent sign and null.
+            private const int  ECVT_MAX_BUFFER_SIZE = 357;  
+            private const int  ECVT_MAX_COUNT_SIZE  = 348; // max digits count for floating numbers
+
+            internal static unsafe void _ecvt_s(byte* buffer, int sizeInBytes, double value, int count, int* dec, int* sign)
```

Comment:
got it. I was thinking to keep _ecvt_s as stand alone method just in case it can be used in different places. but I see DoubleToNumber is enough anyway for such scenario. I'll update the code as you suggested.